lib/_element.js

```diff
@@ -344,6 +352,70 @@ function element () {
     }))
   }
 
+  Element.prototype.compareDocumentPosition = function (other) {
+    function getFirstNodeByOrder (nodes, nodeOne, nodeTwo) {
+      return nodes.reduce(function (result, node) {
+        return result !== false
+          ? result
+          : node === nodeOne
+            ? nodeOne
+            : node === nodeTwo
+              ? nodeTwo
+              : node.childNodes
+                ? getFirstNodeByOrder(node.childNodes, nodeOne, nodeTwo)
+                : false
+      }, false)
+    }
+
+    function isSourceContainTarget (source, target) {
+      while (target.parentNode) {
+        target = target.parentNode
+        if (target === source) {
+          return true
+        }
+      }
+      return false
+    }
+
+    function hasContaining (node, other) {
+      return isSourceContainTarget(node, other)
+        ? Element.DOCUMENT_POSITION_CONTAINED_BY + Element.DOCUMENT_POSITION_FOLLOWING
+        : isSourceContainTarget(other, node)
+          ? Element.DOCUMENT_POSITION_CONTAINS + Element.DOCUMENT_POSITION_PRECEDING
+          : false
```

Comment:
Pretty sure these shouldn't be added together, it's a [bitmask](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators) according to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Node/compareDocumentPosition). So these values won't be correct.

Reply:
I don't agree.

The bit mask is needed to be able to put several values into one value. According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Node/compareDocumentPosition) constants:

| Name | Value | Bitmask |
|-------|-------|---------|
|DOCUMENT_POSITION_DISCONNECTED| 1| 0b00001|
|DOCUMENT_POSITION_PRECEDING| 2| 0b00010|
| DOCUMENT_POSITION_FOLLOWING | 4| 0b00100|
| DOCUMENT_POSITION_CONTAINS | 8| 0b01000|
| DOCUMENT_POSITION_CONTAINED_BY | 16| 0b10000|

are needed to extract the desired "sub-values":

```javascript
console.log(0b01010 & Node.DOCUMENT_POSITION_PRECEDING)
// -> 2 === Node.DOCUMENT_POSITION_PRECEDING
```

As for the correctness of entering several values by the compareDocumentPosition method, I used the example from the tests and ran them in the browser console:

```javascript
/*
 * DOM-structure
 * <documentEl>
 *   <parentEl>
 *     <siblingOne />
 *     <siblingTwo />
 *   </parentEl>
 *   <otherEl>
 *     <siblingThree />
 *   </otherEl>
 * </documentEl>
 * ---
 * <otherDocumentEl />
 */
var documentEl = document.createElement('div')
var parentEl = document.createElement('div')
var siblingOne = document.createElement('div')
var siblingTwo = document.createElement('div')
var otherEl = document.createElement('div')
var siblingThree = document.createElement('div')

documentEl.appendChild(parentEl)
documentEl.appendChild(otherEl)
parentEl.appendChild(siblingOne)
parentEl.appendChild(siblingTwo)
otherEl.appendChild(siblingThree)

siblingTwo.compareDocumentPosition(parentEl).toString(2)
// -> "1010"
```

Reply:
Ah okay! Thanks for the explanation, I get it now :smile: